test/release/examples/primers/errorHandling.chpl

```diff
@@ -0,0 +1,143 @@
+// Error Handling
+//
+// This primer is a short tutorial on Chapel error handling. A detailed
+// outline can be found in the :ref:`readme-errorHandling` technote.
+//
+
+/*
+  .. _primers-errorHandling-handling:
+
+  Handling Errors
+  ---------------
+
+  Standard library functions that used to halt now throw errors for
+  developers to handle as they see fit. This is especially useful for
+  recovering from errors in :mod:`IO`.
+ */
+
+use IO;
+
+const f1 = "fake.txt";
+const f2 = "still_fake.txt";
+
+var f: file;
+
+  // halts on error
+  // f = try! open(f1, iomode.r);
+
+try {
+  f = open(f1, iomode.r);
+} catch {
+  writeln("an error occurred");
+}
+
+try {
+  f = open(f1, iomode.r);
+} catch e: FileNotFoundError {
+  writeln(f1 + " does not exist");
+  try! {
+    f = open(f2, iomode.r);
+  } catch e: FileNotFoundError {
+    writeln(f2 + " does not exist");
+  } // halts if a different error is returned
+} catch {
+  writeln("unknown error");
+}
+
+/*
+  .. _primers-errorHandling-throwing:
+
+  Throwing Errors
+  ---------------
+
+  Developers can also write their own throwing functions, throwing and
+  consuming errors. While there are built-in :class:`SysError.SystemError`
+  types, developers can also create their own hierarchies.
+ */
+
+class EmptyFilenameError : Error { }
+
+proc checkFilename(f_name: string) throws {
+  if f_name.isEmptyString() then
+    throw new EmptyFilenameError();
+}
+
+proc openFilename(f_name: string) throws {
+  var f: file;
+
+  try {
+    f = open(f, iomode.r);
+  } catch e: FileNotFoundError {
+    writeln(f + " does not exist");
```

Comment:
Maybe make it start the writeln with "Warning " ? Also it would be more idomatic to use comma not + between the writeln portions.